yt_dlp/postprocessor/ffmpeg.py

```diff
@@ -659,12 +661,35 @@ def run(self, info):
 
         temp_filename = prepend_extension(filename, 'temp')
         self.to_screen('Embedding subtitles in "%s"' % filename)
-        self.run_ffmpeg_multiple_files(input_files, temp_filename, opts)
-        os.replace(temp_filename, filename)
+        if info['ext'] in self.SUPPORTS_LYRICS:
+            self.embed_lyrics(info['filepath'], sub_dict=info['requested_subtitles'])
+        else:
+            self.run_ffmpeg_multiple_files(input_files, temp_filename, opts)
+            os.replace(temp_filename, filename)
 
         files_to_delete = [] if self._already_have_subtitle else sub_filenames
         return files_to_delete, info
 
+    def embed_lyrics(self, audio_file, sub_dict):
+        if len(sub_dict) > 1:
+            self.report_warning('More than one subtitle file found. Your media player will likely be unable to display all of them.')
+
+        if sub_dict[list(sub_dict.keys())[0]]['ext'] != 'lrc':
```

Comment:
```suggestion
        if not all(sub['ext'] == 'lrc' for sub in sub_dict.values()):
```
we want to check that all subs are lrc format before we can embed

Reply:
`--convert-subs lrc` convert all subs to `lrc` format

else all are `srt ` 

I think only first index is sufficient enough but I will do as you say